### ISSUE TITLE
Critical - intermittent loss of data submitted during the final step of app - per goto meeting

### ISSUE THREAD

**Issue by transystems** [TransystemsLLC/ContactApp-4.6.1#10]
This is the insert of null issue after data has been saved. This was demonstrated during the GoToMeeting session with Tomas.

One more note, when you complete the application, you can click Dashboard to go back to your list of submitted apps, etc. When the condition occurs that prevents the data save, you dashboard screen, which is the next action, is blank. Your name is till logged in at the upper right menu but your data is not processed. It is as if the applicant key is lost for a transaction.

Upon refresh, all data is displayed. 

As you can imagine, this is a critical issue.

**Comment by TomasPDL** [comment]
As we're looking into why the tables are wiping, we're also looking into why the tables are not auto-filling correctly. One of the problems we noticed is that the application is attempting to populate data from the last submitted application. If the last application was submitted with the information nulling error we encountered, then it would stand that it would be pulling in the wrong information back from that application.

As such I went ahead and changed the code so that it will pull from the last submitted _and valid_ application. Being able to ensure that our applications keep auto-filling will help us to continue testing.

We've been able to ensure that it's pulling in information, but are now running into problems with having that information copy over. Progress has been made and we'll continue digging tomorrow.

**Comment by TomasPDL** [comment]
Kevin and I are currently creating a logging system for the code so that we can get some sort of in depth tracking of the errors we're seeing so that we can address what is actually going wrong.

**Comment by transystems** [comment]
I got thinking about your comment…
I am sure you know but… There is currently no activity on either site. Any activity will need to be generated manually.

Dan

**Comment by TomasPDL** [comment]
Right- we can access the application by skipping directly to the page in the URL and then running the application ourselves. We'll be running a couple of apps, tracking our experience and any errors, then comparing them to the logs.

**Comment by TomasPDL** [comment]
The logging code was created and implemented. There are a couple of issues getting certain elements to log correctly, but now we have access to more information regarding the data manipulation than before.

**Comment by TomasPDL** [comment]
I made some commits with new SQL update code in places which overwrites certain old db.Savechanges function. Need to push this across all relevant areas then test.

**Comment by transystems** [comment]
Any thoughts on when this process will be compete?

**Comment by TomasPDL** [comment]
I should have all of the SQL statements in before this weekend, but there's no promise it will run perfectly as is.

**Comment by TomasPDL** [comment]
Update- all of the SQL statements have been made to the application created code. From what I'm seeing the information is being stored correctly and new applications can be created- however we're currently working on a new issue in which the page doesn't advance when you go to create an application. We'll probably need to dive into the angular side of the code for this, I'll keep you posted.
